core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessageDuplicator.java

```diff
@@ -658,7 +658,7 @@ public void request(long n) {
                 return;
             }
 
-            accumulateDemand(n);
+            cumulativeDemand = LongMath.saturatedAdd(cumulativeDemand, n);
```

Comment:
Question) Shouldn’t we take multi-threaded race conditions into account? I might not have a complete understanding of `DownstreamSubscription`’s threading model.

Reply:
I see. If `subscription.request()` should be sequentically called, is it safe to remove the following lines? It seems like `requestedDemand` is always less than `cumulativeDemand` in `doRequestDemand()`. If I'm wrong, may I know when it could happen?
https://github.com/line/armeria/blob/eb7e4512afecb0979e32f17589c7c525aa8662c7/core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessageDuplicator.java#L338-L340

Reply:
> subscription.request() should be sequentically called

This is for one child stream message. 
`processor.requestDemand(cumulativeDemand);` can be called multiple from multiple child stream messages. Let's say there are two child stream messages and the first one request 10 and the second requests 1, then the condition will be true.

Reply:
Thanks for the explanation. Everything makes sense now.